packages/mongo/mongo_driver.js

```diff
@@ -909,30 +888,29 @@ Cursor.prototype.count = function () {
   );
 };
 
-[...ASYNC_CURSOR_METHODS, Symbol.iterator, Symbol.asyncIterator].forEach(methodName => {
-  // count is handled specially since we don't want to create a cursor.
-  // it is still included in ASYNC_CURSOR_METHODS because we still want an async version of it to exist.
-  if (methodName === 'count') {
-    return
-  }
-  Cursor.prototype[methodName] = function (...args) {
-    const cursor = setupSynchronousCursor(this, methodName);
-    return cursor[methodName](...args);
-  };
-
-  // These methods are handled separately.
-  if (methodName === Symbol.iterator || methodName === Symbol.asyncIterator) {
-    return;
-  }
-
-  const methodNameAsync = getAsyncMethodName(methodName);
-  Cursor.prototype[methodNameAsync] = function (...args) {
-    try {
-      return Promise.resolve(this[methodName](...args));
-    } catch (error) {
-      return Promise.reject(error);
+// We don't handle the count method here.
+[...CURSOR_METHODS, Symbol.asyncIterator].filter(method => method !== CURSOR_METHODS[0]).forEach(function(method) {
```

Comment:
```
method !== CURSOR_METHODS[0]
```

If, in the future, we forget about these rules and add something at the top of this array, this code won't work as expected. 

It's best to use the `if` expression here.